core/ir/x86/decode.c

```diff
@@ -2410,6 +2410,14 @@ decode_get_tuple_type_input_size(const instr_info_t *info, decode_info_t *di)
         di->input_size = OPSZ_NA;
 }
 
+bool
+decode_category(int opcode, instr_t *instr)
+{
+    int cat = category_by_opcode[opcode];
```

Comment:
ok. Right now we could add OP_mov_ld in both (load and move)

Reply:
Actually I believe a whole bunch of opcodes are in the same boat: all the TYPE_E operands can be a register or base-disp memory.  So that includes all the regular arithmetic (ADD, SUB, etc.).  Maybe mark them all as the INT or MATH type and then have the code always check for that type (or all types?) whether it reads or writes memory and if so it adds the LOAD or STORE category or whatever the name was for that?

Reply:
As I understand, currently it is in instr_is_mov.

Reply:
@derekbruening could we add all instructions from instr_is_mov to INT | MOVE right now?